---
title: "digestR quickstart"
author: "Andrew D. Steen"
date: "`r Sys.Date()`"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{Vignette Title}
  %\VignetteEngine{knitr::rmarkdown}
  %\usepackage[utf8]{inputenc}
---

# Introduction

**digestR** will eventually consist of 3 key parts:

* a database of protein abundances in environmental samples
* a mapping of pfam codes to hypothesized protein function
* a set of R tools usfeul for identifying spatial trends of gene abundance in the environment. 

Right now, all three tools are embryonic. 

## Database

_I will insert some info about our database as it stands._

## Map

_I will insert some info about our mapping as it stands._

## Tools

The main tool right now looks for pfams or mapped gene functions that correlate with an independent variable such as depth. 

Later, **digestR** will will include mapping tools and more sophisticated algorithms to identify real trends.

# Quickstart

## Importing data

First, **import your data**. Metagenomic data should as a .csv file with a column for each pfam, and a column for each sample containing the number of reads of each pfam. The example dataset includes samples from the Peru Margin named `pm30`, `pm70`, and `pm91`, among others, so the first few columns and rows look like this:

pfam      | pm30    | pm70    | pm91  
----------|---------|---------|--------
pfam00557 | 97      |    16   |  17   
pfam01546 | 58      |    23   |  35
pfam01435 | 33      |     6   |   9

In order to associate the sample names with sample metadata, a sample key is required. The sample key is another .csv file with a column for sample names `sample` and a column for each metadata variable. Minimally, sample data must include latitude `lat`, longitude `lon`, and `depth` (_depth should be in meters; I haven't specified a lat/lon format yet. What's the ISO specification?_), but an arbitrary number of metadata columns (whatever makes sense in the context of your dataset.) The key must also include the **total number of reads** for each metagenome so that the read abundance can be normalized.  

The sample key should be formatted as follows:

sample | lat  | lon   | depth | optional.1 | optional.2 | ...
-------|------|-------|-------|------------|------------|----
pm30   | 05.4 | -123.1| 30    | 17         | 122        | ...
pm70   | 05.4 | -123.1| 70    | 42         | 99         | ...
pm91   | 05.4 | -123.1| 91    | 9          | 504        | ...

The package also uses a `map` to associate pfam codes with hypothesized functions. The package automatically includes a map, but at the moment it is too embryonic to rely on. 

```{r, message=FALSE, warning=FALSE}
library(digestR)

d <- read_digestR()
```

This creates a long-format ('tidy') data frame with the 



Vignettes are long form documentation commonly included in packages. Because they are part of the distribution of the package, they need to be as compact as possible. The `html_vignette` output type provides a custom style sheet (and tweaks some options) to ensure that the resulting html is as small as possible. The `html_vignette` format:

- Never uses retina figures
- Has a smaller default figure size
- Uses a custom CSS stylesheet instead of the default Twitter Bootstrap style

## Vignette Info

Note the various macros within the `vignette` setion of the metadata block above. These are required in order to instruct R how to build the vignette. Note that you should change the `title` field and the `\VignetteIndexEntry` to match the title of your vignette.

## Styles

The `html_vignette` template includes a basic CSS theme. To override this theme you can specify your own CSS in the document metadata as follows:

    output: 
      rmarkdown::html_vignette:
        css: mystyles.css

## Figures

The figure sizes have been customised so that you can easily put two images side-by-side. 

```{r, fig.show='hold'}
plot(1:10)
plot(10:1)
```

You can enable figure captions by `fig_caption: yes` in YAML:

    output:
      rmarkdown::html_vignette:
        fig_caption: yes

Then you can use the chunk option `fig.cap = "Your figure caption."` in **knitr**.

## More Examples

You can write math expressions, e.g. $Y = X\beta + \epsilon$, footnotes^[A footnote here.], and tables, e.g. using `knitr::kable()`.

```{r, echo=FALSE, results='asis'}
knitr::kable(head(mtcars, 10))
```

Also a quote using `>`:

> "He who gives up [code] safety for [code] speed deserves neither."
([via](https://twitter.com/hadleywickham/status/504368538874703872))
